cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -744,6 +786,11 @@ static PyObject *get_precision(PyObject *self, PyObject *args)
 {
     COCOTB_UNUSED(self);
     COCOTB_UNUSED(args);
+
+    if (!simulator_is_active) {
+        return PyLong_FromLong(-15);  // preserves old behavior
```

Comment:
It's probably worth emitting a warning here that no simulator is attached:
```suggestion
        if (PyErr_WarnEx(PyExc_RuntimeWarning, "something about the simulator not actually being attached, assuming whatever precision this is", 1) < 0) {
            return NULL;
        }
        return PyLong_FromLong(-15);  // preserves old behavior
```